### PR TITLE
Add additional tests when interpolating from cell centroids to face c…

### PR DESCRIPTION
**Pull request by asalmgren**
…entroids --

when the face centroid is exactly at the center of the face we need an additional
test to know which way we can look transversely and not hit covered cells.

## Checklist

The proposed changes:
- [X ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [X ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Src/EB/AMReX_EBMultiFabUtil_2D_C.H [hunk 1]
```diff
@@ -415,7 +415,15 @@ void eb_interp_centroid2facecent_x (Box const& ubx,
       else 
       {
           int ii = i - 1;
-          int jj = (fcx(i,j,k) < 0.0) ? j - 1 : j + 1;
+          // We must add an additional test to avoid the case where fcx is very close to zero, but (j-1) or (j+1)
+          //    might be covered cells -- this can happen when the EB is exactly aligned with the horizontal grid lines
+          int jj;
+          if (std::abs(fcx(i,j,k)) > 1.e-8)
```
Comment (WeiqunZhang) [inline Src/EB/AMReX_EBMultiFabUtil_2D_C.H]:
```suggestion
          if (amrex::Math::abs(fcx(i,j,k)) > 1.e-8)
```

**Review by WeiqunZhang — APPROVED** [review]
